tests/app/main/views/test_api_integration.py

```diff
@@ -107,25 +107,33 @@ def test_should_create_api_key_with_type_normal(
     fake_uuid,
     mocker,
 ):
-    post = mocker.patch("app.notify_client.api_key_api_client.ApiKeyApiClient.post", return_value={"data": fake_uuid})
+    mock_api_key_post = mocker.patch("app.notify_client.api_key_api_client.ApiKeyApiClient.post", return_value=None)
+    mock_create_admin_action = mocker.patch("app.admin_actions_api_client.create_admin_action", return_value=None)
 
-    page = client_request.post(
+    key_name = "Some default key name 1/2"
+    client_request.post(
         "main.create_api_key",
         service_id=SERVICE_ONE_ID,
-        _data={"key_name": "Some default key name 1/2", "key_type": "normal"},
-        _expected_status=200,
+        _data={"key_name": key_name, "key_type": "normal"},
+        _expected_redirect=url_for("main.api_keys", service_id=SERVICE_ONE_ID),
     )
 
-    assert page.select_one("span.copy-to-clipboard__value").text == (
-        # The text should be exactly this, with no leading or trailing whitespace
-        f"some_default_key_name_12-{SERVICE_ONE_ID}-{fake_uuid}"
-    )
+    # TODO: How do we assert the flash in a redirect?
```

Comment:
Not sure, but we should definitely find out.  You could mock the flash function perhaps, but I suspect there's a better way